lib/OperationHandler.hpp

```diff
@@ -121,6 +124,9 @@ class OperationHandler: private INotificationHandler
     /// 5) Session ID is expired, prevent the disconnection in the guest
     std::atomic<uint64_t> m_sessionId{};
 
+    /// @brief Set of interfaces currently executing a scan
+    std::set<GUID> m_scanningInterfaces;
```

Comment:
On small collections (and I never expect to see more than 10 items here), `std::set` tends to have less overhead.

Reply:
I agree, though my comment is more about correctness since performance shouldn't matter in this instance.

When I see a `std::set`, I know the elements are _ordered_, so I immediately wonder why the interfaces here needed to be ordered in a particular way and what that means. Taking context into account (set of interfaces that are currently scanning), it seems reasonable that their ordering might actually matter (eg. interfaces that were added first must be processed first). Had this been `std::unordered_set`, it's immediately clear that there is no order and I don't need to think about it.

Anyway, this is more of a philosophical thing and not at all functional, so feel free to ignore.

Reply:
I see your point (and sort of agree with the philosophie behind it). It's unfortunate that the name refers to a different concept instead of a different implementation (like hash_set).